streams/src/main/java/org/apache/kafka/streams/kstream/internals/NamedInternal.java

```diff
@@ -57,40 +59,28 @@ public String name() {
     public NamedInternal withName(final String name) {
         return new NamedInternal(name);
     }
-    
+
     String suffixWithOrElseGet(final String suffix, final String other) {
-        if (name != null) {
-            return name + suffix;
-        } else {
-            return other;
-        }
+        return Optional.ofNullable(name)
+                // Re-validate generated name as suffixed string could be too large.
+                .map(x -> withName(name + suffix).name)
+                .orElse(other);
     }
 
     String suffixWithOrElseGet(final String suffix, final InternalNameProvider provider, final String prefix) {
         // We actually do not need to generate processor names for operation if a name is specified.
         // But before returning, we still need to burn index for the operation to keep topology backward compatibility.
-        if (name != null) {
-            provider.newProcessorName(prefix);
+        final String defaultName = provider.newProcessorName(prefix);
 
-            final String suffixed = name + suffix;
-            // Re-validate generated name as suffixed string could be too large.
-            Named.validate(suffixed);
-
-            return suffixed;
-        } else {
-            return provider.newProcessorName(prefix);
-        }
+        return suffixWithOrElseGet(suffix, defaultName);
     }
 
     String orElseGenerateWithPrefix(final InternalNameProvider provider, final String prefix) {
         // We actually do not need to generate processor names for operation if a name is specified.
         // But before returning, we still need to burn index for the operation to keep topology backward compatibility.
```

Comment:
May I ask a question?
 
Is Backward compatibility still in the current version?
I see the hint in other test

<pre><code>
SuppressTopologyTest.shouldApplyNameToSuppressionNode {
        ...
        .suppress(untilTimeLimit(Duration.ofSeconds(1), unbounded()).withName("asdf"))
        ....
       // <del>without</del>(with) the name,  <b>the suppression node does not increment the topology index </b>
        assertThat(namedNodeTopology, is(NAMED_INTERMEDIATE_TOPOLOGY));
}

Reply:
The NAMED_INTERMEDIATE_TOPOLOGY
<pre><code>
Topologies:
   Sub-topology: 0
    <b>Source: KSTREAM-SOURCE-0000000000 (topics: [input])</b>
      --> KSTREAM-AGGREGATE-0000000002
    <b>Processor: KSTREAM-AGGREGATE-0000000002 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000001])</b>
      --> asdf
      <-- KSTREAM-SOURCE-0000000000
    <b>Processor: asdf (stores: [asdf-store])</b>
      --> KTABLE-TOSTREAM-0000000003
      <-- KSTREAM-AGGREGATE-0000000002
    <b>Processor: KTABLE-TOSTREAM-0000000003 (stores: [])</b>
      --> KSTREAM-SINK-0000000004
      <-- asdf
    <b>Sink: KSTREAM-SINK-0000000004 (topic: output)</b>
      <-- KTABLE-TOSTREAM-0000000003
</code></pre>